### PR TITLE
dbeaver/pro#2365 read global properties if new connection

### PR DESCRIPTION
**Pull request by Destrolaric**
Fixes the situation when global properties were changed, and no settings page was previously loaded, leading to settings not being applied
Criteria:
* **No changes should apply retroactively to existing connections**
* New connections should use global settings even if the advanced settings page weren't loaded previously

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java [hunk 1]
```diff
@@ -16,6 +16,17 @@
...
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.ZoneId;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreDataSource.java]: Please change order of imports back.

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java [hunk 2]
```diff
@@ -16,6 +16,8 @@
  */
 package org.jkiss.dbeaver.ext.postgresql.ui;
 
+import java.util.Locale;
+
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java]: Why did you move it at the beginning of imports?

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java [hunk 3]
```diff
@@ -297,6 +301,24 @@ public void saveSettings(DBPDataSourceContainer dataSource) {
...
+    private static void applyGlobalProperties(DBPConnectionConfiguration connectionInfo) {
+        // Additional page can be not loaded at all, if this is the case just to be sure we need to apply global settings
+        String[] propertiesToCheck = {
+            PostgreConstants.PROP_SHOW_NON_DEFAULT_DB,
+            PostgreConstants.PROP_SHOW_UNAVAILABLE_DB,
+            PostgreConstants.PROP_SHOW_TEMPLATES_DB,
+            PostgreConstants.PROP_READ_ALL_DATA_TYPES,
+            PostgreConstants.PROP_READ_KEYS_WITH_COLUMNS,
+            PostgreConstants.PROP_SHOW_DATABASE_STATISTICS
+        };
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java:313]: I don't like an idea of enumerating properties.

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java (hunk 2 above)
Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java]: Fixed

File: plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java (hunk 3 above)
Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/PostgreConnectionPage.java:313]: The idea is that we don't need to set properties from additional page. Setting all properties looks really bad (we will push a lot of useless stuff in all connections and we won't be able to change defaults after it)

**Review by serge-rider — CHANGES_REQUESTED** [review]
